javascript/ql/experimental/adaptivethreatmodeling/lib/experimental/adaptivethreatmodeling/FunctionBodyFeatures.qll

```diff
@@ -127,22 +127,29 @@ ASTNode getAnASTNodeWithAFeature(Function f) {
   result = getAnASTNodeToFeaturize(f)
 }
 
+int getNumCharsInFunction(Function f) {
+    result = strictsum(int i |
+      exists(ASTNode node | node = getAnASTNodeWithAFeature(f) and i = getTokenizedAstNode(node).length()) |
+      i
+    )
```

Comment:
This sums all integers i such that there exists a tokenized AST node of that length.  This will give a result of 2 instead of 4 if we have for instance two AST nodes of length 2.  Instead we should sum over the AST nodes, i.e. `strictsum(ASTNode node | node = getAnASTNodeWithAFeature(f) | getTokenizedAstNode(node).length())`.

Reply:
Oof .. thanks for the explanation. Will fix.